### PR TITLE
Convert stats to phasers

### PR DESCRIPTION
**Pull request by rr-**
#### Checklist

- [x] I have read the [coding conventions](https://github.com/LostArtefacts/TR1X/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] I have added a changelog entry about what my pull request accomplishes, or it is an internal change

#### Description

Another PR in our quest to get rid of multiple event loops.

### CODE REVIEW

**Comment by walkawayy** [comment]
Getting a compile error. Code looks good to me otherwise. I can't gameplay test though.

```
FAILED: TR1X.exe.p/src_game_gameflow.c.obj 
/usr/bin/i686-w64-mingw32-gcc -ITR1X.exe.p -I. -I../.. -I../../src -I/ext//include -I/ext/include -I/ext/include/SDL2 -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -O0 -g -Wno-unused -D_GNU_SOURCE -DMESON_BUILD -Dmain=SDL_main -MD -MQ TR1X.exe.p/src_game_gameflow.c.obj -MF TR1X.exe.p/src_game_gameflow.c.obj.d -o TR1X.exe.p/src_game_gameflow.c.obj -c ../../src/game/gameflow.c
../../src/game/gameflow.c: In function ‘GameFlow_InterpretSequence’:
../../src/game/gameflow.c:1313:13: error: a label can only be part of a statement and a declaration is not a statement
 1313 |             PHASE_STATS_DATA phase_args = {
      |             ^~~~~~~~~~~~~~~~
```

File: src/game/gameflow.c [hunk 1]
```diff
@@ -1310,15 +1310,31 @@ GameFlow_InterpretSequence(int32_t level_num, GAMEFLOW_LEVEL_TYPE level_type)
             break;
 
         case GFS_LEVEL_STATS:
-            Stats_Show((int32_t)(intptr_t)seq->data);
+            PHASE_STATS_DATA phase_args = {
```
Comment (walkawayy) [inline src/game/gameflow.c:1313]: Add parenthesis around this statement to fix compile fail.

**Review by walkawayy — CHANGES_REQUESTED** [review]
All testing looks good once the compilation is fixed.

File: src/game/gameflow.c (hunk 1 above)
Comment (rr-) [reply src/game/gameflow.c:1313]: Addressed

**Review by walkawayy — APPROVED** [review]
LGTM

**Pull request merged**